docker-environment-driver/src/main/resources/isolatedDockerConfiguration/init.groovy

```diff
@@ -0,0 +1,9 @@
+import hudson.model.*;
+import jenkins.model.*;
+
+
+Thread.start {
+      sleep 10000
+      println "--> setting agent port for jnlp"
+      Jenkins.instance.setSlaveAgentPort(50000)
```

Comment:
Perhaps it would be a bit simpler to use Jenkins slave.jar file. It should do it automatically without involving custom groovy script here.